src/NuGet.Services.AzureSearch/SearchService/SearchTextBuilder.cs

```diff
@@ -53,48 +53,73 @@ public string V3Search(V3SearchRequest request)
             return GetLuceneQuery(request.Query);
         }
 
+        public string Autocomplete(AutocompleteRequest request)
+        {
+            if (string.IsNullOrWhiteSpace(request.Query))
+            {
+                return MatchAllDocumentsQuery;
+            }
+
+            // Generate a query on package ids. This will be a prefix search
+            // if we are autocompleting package ids.
+            var builder = new AzureSearchQueryBuilder();
+
+            builder.AddScopedTerms(
+                IndexFields.PackageId,
```

Comment:
AddScopedTerms can throw, what will be the experience in this case?

Reply:
That's a very good question. This will throw an uncaught exception and result in a HTTP 500 Internal Error. I'll fix this

Reply:
This should be fixed now